poetry/core/version/pep440/version.py

```diff
@@ -46,7 +47,21 @@ def __post_init__(self):
 
         object.__setattr__(self, "_compare_key", self._make_compare_key())
 
-    def _make_compare_key(self):
+    def _make_compare_key(
```

Comment:
Wow, this is a complicated type. Is there any way to clean this up with NewType, a refactor with an object, or similar?

Reply:
I was happy, that PyCharm was doing this for me :laughing: IMO refactoring should be done in another PR, because the purpose of this PR is to enforce type hints in the future.

Reply:
That is a brutal type signature :)

Reply:
I've tried to understand the code and IMO the signature can be written more easily. See code change.